pkg/labels/parse.go

```diff
@@ -30,27 +35,58 @@ var (
 	}
 )
 
+// ParseMatchers parses a comma-separated list of Matchers. A leading '{' and/or
+// a trailing '}' is optional and will be trimmed before further
+// parsing. Individual Matchers are separated by commas outside of quoted parts
+// of the input string. Those commas may be sorrounded by whitespace. Parts of the
+// string inside unescaped double quotes ('"…"') are considered quoted (and
+// commas don't act as separators there). If double quotes are escaped with a
+// single backslash ('\"'), they are ignored for the purpose of identifying
+// quoted parts of the input string. If the input string, after trimming the
+// optional trailing '}', ends with a comma, followed by optional whitespace,
+// this comma and whitespace will be trimmed.
+//
+// Examples for valid input strings:
+//   {foo = "bar", dings != "bums", }
+//   foo=bar,dings!=bums
+//   {quote="She said: \"Hi, ladies! That's gender-neutral…\""}
+//   statuscode=~"5.."
+//
+// See ParseMatcher for details how an individual Matcher is parsed.
 func ParseMatchers(s string) ([]*Matcher, error) {
 	matchers := []*Matcher{}
 	s = strings.TrimPrefix(s, "{")
 	s = strings.TrimSuffix(s, "}")
 
-	var insideQuotes bool
-	var token string
-	var tokens []string
+	var (
+		insideQuotes bool
+		escaped      bool
+		token        strings.Builder
```

Comment:
Just a doubt : Why did we change token's data type ?

Reply:
Previously, the infamous string concatenation with quadratic runtime was used: `token += string(r)` (line 47 prior to this PR). `strings.Builder` is the straightforward way to avoid it.